stdlib/LinearAlgebra/src/qr.jl

```diff
@@ -567,14 +567,18 @@ size(F::Union{QR,QRCompactWY,QRPivoted}) = size(getfield(F, :factors))
 size(Q::AbstractQ, dim::Integer) = size(getfield(Q, :factors), dim == 2 ? 1 : dim)
 size(Q::AbstractQ) = size(Q, 1), size(Q, 2)
 
-function getindex(Q::AbstractQ, i::Integer, j::Integer)
-    x = zeros(eltype(Q), size(Q, 1))
-    x[i] = 1
+copy(Q::AbstractQ) = I(size(Q, 1)) * Q
```

Comment:
What was wrong about `Matrix(Q)`? The fact that it doesn't necessarily produce a square matrix? Then I'd actually suggest
```suggestion
copy(Q::AbstractQ{T}) where {T} = lmul!(Q, Matrix{T}(I, size(Q)))
```
The fallback method of `copy(Q)` is superslow for larger `Q`. My suggestion follows our definition of `Matrix(Q)`, but makes sure that you get a square matrix, just as `copy(Q)` gives right now.